test/generators/rubocop_install_generator_test.rb

```diff
@@ -15,53 +15,77 @@ def destination_root
     app_path
   end
 
-  def test_should_install_rubocop_successfully
-    Dir.chdir(app_path) do
-      quietly { run_generator }
+  class BaseConfigurationsTest < self
+    def test_should_install_rubocop_successfully
+      Dir.chdir(app_path) do
+        quietly { run_generator }
 
-      assert_file "Gemfile" do |content|
-        assert_match(/rubocop/, content)
-        assert_match(/rubocop-rails/, content)
-        assert_match(/rubocop-performance/, content)
+        assert_file "Gemfile" do |content|
+          assert_match(/rubocop/, content)
+          assert_match(/rubocop-rails/, content)
+          assert_match(/rubocop-performance/, content)
+          assert_match(/rubocop-rake/, content)
+        end
+
+        assert_file ".rubocop.yml" do |content|
+          assert_match("2.7.1", content)
+          assert_match("AllCops", content)
+        end
       end
+    end
 
-      assert_file ".rubocop.yml" do |content|
-        assert_match("2.7.1", content)
-        assert_match("AllCops", content)
+    def test_should_skip_adding_rubocop_rules
+      Dir.chdir(app_path) do
+        quietly { run_generator [destination_root, "--skip_adding_rubocop_rules"] }
+
+        assert_file ".rubocop.yml" do |content|
+          assert_no_match("AllCops", content)
+          assert_no_match("2.7.1", content)
+        end
+      end
+    end
+
+    def test_should_add_target_ruby_version
+      Dir.chdir(app_path) do
+        quietly { run_generator [destination_root, "--ruby_version=2.6.3"] }
+
+        assert_file ".rubocop.yml" do |content|
+          assert_match("AllCops", content)
+          assert_match("2.6.3", content)
+        end
       end
     end
   end
 
-  def test_should_skip_adding_rubocop_rules
-    Dir.chdir(app_path) do
-      quietly { run_generator [destination_root, "--skip_adding_rubocop_rules"] }
+  class TestFrameworkGemTest < self
```

Comment:
@coolprobn I prefer writing separate test cases than adding nested classes. Would you like to address this in next PR?

Reply:
I am new to writing automated tests so I don't have much thoughts on best practices. I will look forward to your guidance.

I agree on the part that it will be hard to understand for first-time contributors.

When you say separate test cases, do we add two different class to the same file or to their separate file too?

My thoughts while working on this was, "how do I group tests together which are for the same feature", in rspec we could do that with "context" and with Ruby way I thought this would be suitable. Maybe we could try using modules like suggested here? https://twitter.com/GregMolnar/status/1642535461947146240?s=20

Reply:
> When you say separate test cases, do we add two different class to the same file or to their separate file too?

If I had to write it, I wouldn't worry about context here instead add a more detailed test case description. You can also extract them to other test files for simplicity. 

> Maybe we could try using modules like suggested here?
Module could be useful if you plan to re-run those test cases in other generators. If not, I don't think extracting to the module adds more value. 

Example: In the Rails repo, you can see they have the `SharedGeneratorTests` module included in few test cases: https://github.com/rails/rails/blob/6c43ebc220428ce9fc9569c2e5df90a38a4fc4e4/railties/test/generators/shared_generator_tests.rb

Reply:
Agreed, for making tests simple and readable it will be a good idea if we don't use nested classes.

I will extract these tests to two separate files if so like you have suggested.

Reply:
@abhaynikam I have separated test classes and pushed the changes.